src/celeritas/optical/surface/SurfacePhysicsView.hh

```diff
@@ -159,9 +164,11 @@ template<class T>
 CELER_FUNCTION T SurfacePhysicsView::oriented_map(
     ItemMap<SurfaceTrackPosition, T> const& map, SurfaceTrackPosition pos) const
 {
-    auto index = orientation_ == SubsurfaceDirection::reverse
-                     ? SurfaceTrackPosition{(map.size() - 1) - pos.get()}
-                     : pos;
+    CELER_EXPECT(pos);
+    auto index
+        = orientation_ == SubsurfaceDirection::reverse
+              ? SurfaceTrackPosition{map.size() - 1 - pos.unchecked_get()}
```

Comment:
I think some static casts here as well, and also an assertion that map isn't empty?